src/models/Settings.js

```diff
@@ -254,12 +254,15 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, OAuth2Util, config) {
           });
         }
       },
-      // filters the idps passed into the widget to include only the ones we support.
+      // Adjusts the idps passed into the widget based on if they get explicit support
       configuredSocialIdps: {
         deps: ['idps'],
         fn: function (idps) {
-          return _.filter(idps, function (idp) {
-            return _.contains(supportedIdps, idp.type.toLowerCase());
+          return _.map(idps, function (idp) { 
+            if( ! ( idp.type && _.contains(supportedIdps, idp.type.toLowerCase()) )) {
+              idp.type = 'general-idp';
```

Comment:
Am I reading wrong that the logic here is to override `type` to `general-idp` as long as not one of those 4 supported types? Then what's the point allow developer to specify `type`.

Reply:
Largely to have something to hang the future expected features on - custom CSS classes, potential i18n keys, etc, but it could be something that gets dropped.  Easier to have customers keep using something they have than to retroactively add it after dropping it.  @nbarbettini  - any thoughts?

Reply:
After talking with Nate - type is optional and is no longer mentioned for generic IDPs, and I've added a `className` property to allow for custom styling.  i18n will be dealt with later when we have a consistent approach.

Reply:
Revised: type is no longer required.  Unknown types will be treated just like absent types (though we might add some console logging to that effect in a later round).  Type is meaningful only for the existing four hardcoded options.